Features/Lookuptable/testPages/data/lookup_table_page.py

```diff
@@ -0,0 +1,771 @@
+
+    def download_update_7(self, tableid, path):
+        excel = ExcelManager(path)
+        df1 = excel.read_excel(tableid)
+        self.err_upload(path)
+        df2 = excel.read_excel(tableid)
+        return df1, df2
+
+    def download_update_8(self,path,table_id):
+        excel = ExcelManager(path)
+        col = excel.col_size(table_id)
+        excel.write_excel_data(table_id, 1, col + 1, "user 1")
+        excel.write_excel_data(table_id, 1, col + 2, "group 1")
+        excel.write_data(table_id, UserData.data_list1)
+        self.err_upload(path)
+        d1 = excel.read_excel(table_id)
+        print("data1", d1)
+        excel.write_excel_data(table_id, 2, 4, 'kiran')
+        excel.write_excel_data(table_id, 2, 5, '123')
+        self.err_upload(path)
+        d2 = excel.read_excel(table_id)
+        print("data2", d2)
+        return d1, d2
+
+    def replace_existing_table(self, filepath):
+        self.wait_to_click(self.manage_tables_link)
+        self.scroll_to_bottom()
+        self.send_keys(self.upload_table, filepath)
+        self.wait_to_click(self.replace_table)
+        self.wait_to_click(self.upload)
+        time.sleep(20)
+        self.wait_to_click(self.manage_tables_link)
+
+
+    def delete_row_from_table(self, downloadpath, tablename):
+        download_path = self.latest_download_file()
+        excel = ExcelManager(download_path)
+        excel.write_data(tablename, UserData.data_list1)
+        self.err_upload(download_path)
+        self.download1()
+        self.view_lookup_table(tablename)
+        row_count_before = self.rowCount_table(tablename)
+        download_path_1 = self.latest_download_file()
+        excel = ExcelManager(download_path_1)
+        excel.delete_row(tablename, 2)
+        self.replace_existing_table(download_path_1)
+        self.view_lookup_table(tablename)
+        row_count_after = self.rowCount_table(tablename)
+        assert row_count_before > row_count_after
+        print("Row got successfully Deleted")
+
+    def rowCount_table(self, tablename):
+        self.view_lookup_table(tablename)
+        text = self.get_text(self.rowcount)
+        rowcount = text[18:20].strip()
+        return rowcount
+
+    def update_delete_field(self,download_path,tablename):
+        excel = ExcelManager(download_path)
+        excel.upload_to_path(tablename, UserData.data_list)
+        self.err_upload(download_path)
+        self.view_lookup_table(tablename)
+        row_count_before = self.rowCount_table(tablename)
+        download_path_1 = self.latest_download_file()
+        excel = ExcelManager(download_path_1)
+        excel.write_excel_data(tablename, 3, 2, 'Y')
+        self.replace_existing_table(download_path_1)
+        self.view_lookup_table(tablename)
+        row_count_after = self.rowCount_table(tablename)
+        assert row_count_before > row_count_after
+        print("Row got successfully Deleted")
+
+    def attribute_2(self,download_path,tablename):
```

Comment:
remove download_path from parameters, as you are using it after defining inside the function.